packages/teams-js/src/internal/utils.ts

```diff
@@ -37,41 +37,32 @@ function validateHostAgainstPattern(pattern: string, host: string): boolean {
   return false;
 }
 
-const validateOriginLogger = getLogger('validateOrigin');
+export const validateOriginLogger = getLogger('validateOrigin');
+
+export function sanitizeAndStrigifyURL(url: URL): string {
+  const stringifiedUrl = JSON.stringify(url);
+  const sanitizedUrl = JSON.parse(stringifiedUrl);
+  return sanitizedUrl;
+}
 
 /**
  * @internal
  * Limited to Microsoft-internal use
  */
-export function validateOrigin(messageOrigin: URL): boolean {
-  // Check whether the url is in the pre-known allowlist or supplied by user
-  if (!isValidHttpsURL(messageOrigin)) {
-    validateOriginLogger(
-      'Origin %s is invalid because it is not using https protocol. Protocol being used: %s',
-      messageOrigin,
-      messageOrigin.protocol,
-    );
-    return false;
-  }
-  const messageOriginHost = messageOrigin.host;
-
+export function fallbackValidateOrigin(messageOriginHost: string): boolean {
   if (validOrigins.some((pattern) => validateHostAgainstPattern(pattern, messageOriginHost))) {
     return true;
   }
+  return false;
+}
 
+export function validateAdditionalOrigins(messageOriginHost: string): boolean {
   for (const domainOrPattern of GlobalVars.additionalValidOrigins) {
     const pattern = domainOrPattern.substring(0, 8) === 'https://' ? domainOrPattern.substring(8) : domainOrPattern;
     if (validateHostAgainstPattern(pattern, messageOriginHost)) {
       return true;
     }
   }
-
-  validateOriginLogger(
```

Comment:
Would a modified version of this logging still be useful here? Seems like it would help to see when messages are getting rejected for this reason

Reply:
This specifically logs that the entire domain validation process has failed. In the new validateOrigin method we call fallbackValidateOrigin and validateAdditionalOrigin it would have to fail all 3 methods to be logged.

I could add additional logging, but messages being rejected would only occur if all 3 fail and I don't want to log failures that succeed later down the line since then essentially every message that goes through the app supplied list will get logged as a failure before succeeding. What I think might be interesting is the case where the service endpoint call fails, but the fallback passes.